### PR TITLE
Issue/handler typing

### PR DESCRIPTION
**Pull request by bartv**

### CODE REVIEW

File: src/inmanta/agent/handler.py [hunk 1]
```diff
@@ -142,7 +153,7 @@ def __init__(self, resource, dry_run=False, action_id=None, logger=None):
         self._created = False
         self._change = const.Change.nochange
 
-        self._changes: Dict[str, Dict[str, str]] = {}
+        self._changes: Dict[str, AttributeStateChange] = {}
```
Comment (wouterdb) [inline src/inmanta/agent/handler.py]: L582  (dryrun execute) goes south with this.

File: src/inmanta/agent/handler.py [hunk 2]
```diff
@@ -77,7 +86,7 @@ class InvalidOperation(Exception):
 
 
 def cache(
```
[the remark this answers is not in the document]
Comment (wouterdb) [reply src/inmanta/agent/handler.py]: (I will push something to this effect)

File: src/inmanta/env.py [hunk 3]
```diff
@@ -84,7 +85,7 @@ def init_env(self):
         self.virtual_python = python_bin
         return True
```
Comment (arnaudsjs) [inline src/inmanta/env.py]: Why is this return statement present?

File: src/inmanta/protocol/endpoints.py [hunk 4]
```diff
@@ -132,10 +132,11 @@ class SessionEndpoint(Endpoint, CallTarget):
         An endpoint for clients that make calls to a server and that receive calls back from the server using long-poll
     """
 
+    _client: "SessionClient"
```
Comment (arnaudsjs) [inline src/inmanta/protocol/endpoints.py]: Shouldn't this be optional?

File: src/inmanta/agent/agent.py [hunk 5]
```diff
@@ -488,7 +518,9 @@ def reload(self, resources, undeployable={}, reason: str = "RELOAD", is_repair=F
         # Start running
         dummy.future.set_result(ResourceActionResult(True, False, False))
 
-    async def mark_deployment_as_finished(self, resource_actions, reason, gid):
+    async def mark_deployment_as_finished(
+        self, resource_actions: Iterable[ResourceAction], reason: str, gid: uuid.UUID
```
[the remark this answers is not in the document]
Comment (bartv) [reply src/inmanta/agent/agent.py]: done

File: src/inmanta/env.py (hunk 3 above)
Comment (bartv) [reply src/inmanta/env.py]: done

File: src/inmanta/agent/io/local.py [hunk 6]
```diff
@@ -446,6 +479,7 @@ def symlink(self, source, target):
         return os.symlink(source, target)
```
[the remark this answers is not in the document]
Comment (bartv) [reply src/inmanta/agent/io/local.py]: done

File: src/inmanta/protocol/endpoints.py (hunk 4 above)
Comment (bartv) [reply src/inmanta/protocol/endpoints.py]: no, because that makes all other code more complex. It is set in start so, should be available.

**Pull request merged**